errors.go

```diff
@@ -50,11 +54,27 @@ var (
 	errNoAvailableCipherSuites           = errors.New("dtls: Connection can not be created, no CipherSuites satisfy this Config")
 	errInvalidClientKeyExchange          = errors.New("dtls: Unable to determine if ClientKeyExchange is a public key or PSK Identity")
 	errNoSupportedEllipticCurves         = errors.New("dtls: Client requested zero or more elliptic curves that are not supported by the server")
-	errConnectTimeout                    = errors.New("dtls: The connection timed out during the handshake")
 	errRequestedButNoSRTPExtension       = errors.New("dtls: SRTP support was requested but server did not respond with use_srtp extension")
 	errClientNoMatchingSRTPProfile       = errors.New("dtls: Server responded with SRTP Profile we do not support")
 	errServerNoMatchingSRTPProfile       = errors.New("dtls: Client requested SRTP but we have no matching profiles")
 	errServerRequiredButNoClientEMS      = errors.New("dtls: Server requires the Extended Master Secret extension, but the client does not support it")
 	errClientRequiredButNoServerEMS      = errors.New("dtls: Client required Extended Master Secret extension, but server does not support it")
 	errInvalidCertificate                = errors.New("dtls: No certificate provided")
+
+	// Wrapped errors
+	errConnectTimeout = &wrappedError{"dtls: The connection timed out during the handshake", context.DeadlineExceeded}
 )
+
+type wrappedError struct {
```

Comment:
Use `fmt.Errorf("...: %w")` instead after Go 1.12 EOL.

Reply:
Perhaps we could use https://godoc.org/golang.org/x/xerrors over our own type?

Reply:
Definitely! I'll switch to use xerrors.